### DISCUSSION TITLE
Batch insert when using identity generator and stateless sessions?

### DISCUSSION THREAD

**Discussion by stucampbell** [nhibernate/nhibernate-core#2662] [General]
NH disables batching for entities that use an identity generator. I understand why this would be the case for stateful sessions but I'm curious to know why this rule still applies to stateless sessions and why there isn't a way to circumvent the rule.

I have tens of thousands of records to insert and I don't need the the ID values returned, yet each row is being inserted with a roundtrip to the database so that the SCOPE_IDENTITY() is returned.
This is taking minutes to execute. What I'd like to be able to do is to implement a way of forcing batching (regardless of the identity generator) in this particular scenario. Can anyone think of a reason why this would be a bad thing to implement in NH?

**Comment by oskarb** [comment]
It does sound like you are asking for the stateless session to have state.
That might be one reason not to do it.

**Comment by hazzik** [comment]
Sounds like a case for `Persist`
